### ISSUE TITLE
[Feature] Add data source as `github_repository_file`

### ISSUE THREAD

**Issue by rrmistry** [integrations/terraform-provider-github#825]
It would be good if there was a data source that can pull the GitHub file's contents.

Ideally a `data` source compliment to [`github_repository_file`](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_file)

Currently doing a workaround using `http` resource calling GitHub REST API:

```terraform
data "http" "logicAppSource" {
  # Get source code for LogicApp definition from specific GitHub repo commit
  url = "https://api.github.com/repos/<ORG>/<REPO>/contents/logicapp_definitions/<APP_NAME>.logicapp.json?ref=${var.github_reference}"

  request_headers = {
    Accept = "application/vnd.github.v3.raw+text"
    Authorization = "token ${var.github_token}"
  }
}

# Usage as below
module "logicapp_<APP_NAME>" {
  source = "github.com/<ORG>/<REPO>"
  definition = data.http.logicAppSource.body # Usage like this
}
```

The GitHub REST API approach works as a workaround but constantly throws Terraform warnings:
```log
╷
│ Warning: Content-Type is not recognized as a text type, got "application/vnd.github.v3.raw+text; charset=utf-8"
│
│   with data.http.logicAppSource,
│   on logicapp_logicAppSource.tf line 1, in data "http" "logicAppSource":
│    1: data "http" "logicAppSource" {
│
│ If the content is binary data, Terraform may not properly handle the contents of the response.
│
│ (and 5 more similar warnings elsewhere)
╵
```

__________

Workaround Terraform versions:
```sh
Terraform v0.15.1
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.62.0
+ provider registry.terraform.io/hashicorp/http v2.1.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
```

**Comment by kclarkey** [comment]
Circling back to this as it appears this feature was implemented back in September and released with version [4.16.0](https://github.com/integrations/terraform-provider-github/tree/v4.16.0), but this issue is still open and I see no mention of the new feature referenced in the changelog. Figured it was worth a mention.

**Comment by jcudit** [comment]
Thanks for circling back 🙇🏾 

Seems closed now.
